contracts/verifiers/UniversalVerifier.sol

```diff
@@ -21,216 +23,234 @@ contract UniversalVerifier is
-    /// @param request The ZKP request data
-    function updateZKPRequest(
-        uint64 requestId,
-        IZKPVerifier.ZKPRequest calldata request
-    ) public onlyOwner {
-        super._updateZKPRequest(requestId, request);
+    /**
+     * @dev Updates a request
+     * @param request The request data
+     */
+    function updateRequest(IVerifier.Request calldata request) public onlyOwner {
+        super._updateRequest(request);
 
-        emit ZKPRequestUpdate(
-            requestId,
+        emit RequestUpdate(
+            request.requestId,
             _msgSender(),
             request.metadata,
             address(request.validator),
-            request.data
+            request.params
         );
     }
 
-    /// @dev Submits a ZKP response and updates proof status
-    /// @param requestId The ID of the ZKP request
-    /// @param inputs The input data for the proof
-    /// @param a The first component of the proof
-    /// @param b The second component of the proof
-    /// @param c The third component of the proof
-    function submitZKPResponse(
-        uint64 requestId,
-        uint256[] memory inputs,
-        uint256[2] memory a,
-        uint256[2][2] memory b,
-        uint256[2] memory c
-    ) public override {
-        super.submitZKPResponse(requestId, inputs, a, b, c);
-        emit ZKPResponseSubmitted(requestId, _msgSender());
+    /**
+     * @dev Sets a multiRequest
+     * @param multiRequest The multiRequest data
+     */
+    function setMultiRequest(
+        IVerifier.MultiRequest calldata multiRequest
+    ) public override checkMultiRequestExistence(multiRequest.multiRequestId, false) {
+        super.setMultiRequest(multiRequest);
+        emit MultiRequestSet(
+            multiRequest.multiRequestId,
+            multiRequest.requestIds,
+            multiRequest.groupIds
+        );
     }
 
     /**
-     * @dev Submits an array of ZKP responses and updates proofs status
-     * @param responses The list of responses including ZKP request ID, ZK proof and metadata
-     * @param crossChainProof The list of cross chain proofs from universal resolver (oracle). This
+     * @dev Submits an array of responses and updates proofs status
+     * @param authResponse Auth responses including auth type and proof
+     * @param responses The list of responses including request ID, proof and metadata for requests
+     * @param crossChainProofs The list of cross chain proofs from universal resolver (oracle). This
      * includes identities and global states.
      */
-    function submitZKPResponseV2(
-        IZKPVerifier.ZKPResponse[] memory responses,
-        bytes memory crossChainProof
+    function submitResponse(
+        AuthResponse memory authResponse,
+        Response[] memory responses,
+        bytes memory crossChainProofs
     ) public override {
-        super.submitZKPResponseV2(responses, crossChainProof);
+        super.submitResponse(authResponse, responses, crossChainProofs);
```

Comment:
Does it check for disabled request and (not) whitelisted validators?

Reply:
Yes, we check it via `_getRequestIfCanBeVerified`, which is defined at Verifier level and overridden at `RequestOwnership` and `ValidatorWhitelist` level.